File: bleep-core/src/main/scala/bleep/UserPaths.scala
package bleep

import coursier.cache.shaded.dirs.ProjectDirectories

import java.nio.file.Path

case class UserPaths(cacheDir: Path, configDir: Path) {
  val bspSocketDir = cacheDir / "bsp-socket"
  val coursierCacheDir = cacheDir / "coursier"
  val configJson = configDir / "config.json"
}

object UserPaths {
  def fromAppDirs: UserPaths = {
    val dirs = ProjectDirectories.from("no", "arktekk", "bleep")
    val cacheDir = Path.of(dirs.cacheDir)
    val configDir = Path.of(dirs.configDir)

    UserPaths(cacheDir, configDir)
  }
}

File: bleepscript/src/main/java/bleepscript/UserPaths.java
package bleepscript;

import java.nio.file.Path;
import java.util.Objects;

public record UserPaths(
    Path cacheDir,
    Path configDir,
    Path bspSocketDir,
    Path resolveCacheDir,
    Path resolveJvmCacheDir,
    Path configYaml) {
  public UserPaths {
    Objects.requireNonNull(cacheDir, "cacheDir");
    Objects.requireNonNull(configDir, "configDir");
    Objects.requireNonNull(bspSocketDir, "bspSocketDir");
    Objects.requireNonNull(resolveCacheDir, "resolveCacheDir");
    Objects.requireNonNull(resolveJvmCacheDir, "resolveJvmCacheDir");
    Objects.requireNonNull(configYaml, "configYaml");
  }
}
